cargo chef cook ignores disabled default features in dependencies

Hi,

I hit an issue where `cargo chef cook --release` ignores the disabling of default features for dependencies, making it impossible to build Rusoto with Rustls.

In a workspace with layout:

```
.
├── backend
│   ├── Cargo.toml
│   └── src
├── Cargo.lock
├── Cargo.toml
├── Dockerfile
├── frontend_yew
│   ├── Cargo.toml
│   ├── index.html
│   ├── main.js
│   ├── pkg
│   ├── README.md
│   └── src
├── LICENSE-APACHE
├── LICENSE-MIT
├── README.md
```

and the backend Cargo.toml:

```toml
[dependencies]
actix-web = "3.2.0"
tracing-actix-web = "0.2.1"
tracing = "0.1.21"
serde_derive = "1.0.117"
serde = "1.0.117"
base-62 = "0.1.1"
uuid = { version = "0.8.1", features = ["v4"] }
tracing-futures = "0.2.4"
anyhow = "1.0.34"
rusoto_core = { version = "0.45.0", default_features=false, features=["rustls"] }
rusoto_s3 = { version = "0.45.0", default_features=false, features=["rustls"] }
tokio = "0.2"
lazy_static = "1.4.0"
tracing-log = "0.1.1"
tracing-subscriber = "0.2.15"
tracing-bunyan-formatter = "0.1.6"
actix-cors = "0.5.1"
```

Running `cargo chef prepare  --recipe-path recipetest.json` generates:
```json

{
  "skeleton": {
    "manifests": [
      {
        "relative_path": "Cargo.toml",
        "contents": "[workspace]\nmembers = [\"backend\"]\n"
      },
      {
        "relative_path": "frontend_yew/Cargo.toml",
        "contents": "bench = []\nbin = []\nexample = []\ntest = []\n\n[dependencies]\nwasm-bindgen = \"0.2.68\"\n\n[dependencies.web-sys]\nfeatures = [\"FormData\"]\noptional = false\nversion = \"0.3.45\"\n\n[dependencies.yew]\nfeatures = [\"services\"]\noptional = false\nversion = \"0.17.4\"\n\n[lib]\nbench = true\ncrate-type = [\"cdylib\", \"rlib\"]\ndoc = true\ndoctest = true\nharness = true\nplugin = false\nproc-macro = false\nrequired-features = []\ntest = true\n\n[package]\nauthors = [\"James McMurray <jamesmcm03@gmail.com>\"]\nautobenches = true\nautobins = true\nautoexamples = true\nautotests = true\ncategories = []\nedition = \"2018\"\nkeywords = []\nname = \"frontend\"\npublish = true\nversion = \"0.1.0\"\n"
      },
      {
        "relative_path": "backend/Cargo.toml",
        "contents": "bench = []\nexample = []\ntest = []\n\n[[bin]]\nbench = true\ndoc = true\ndoctest = true\nedition = \"2018\"\nharness = true\nname = \"elposhigu\"\npath = \"src/main.rs\"\nplugin = false\nproc-macro = false\nrequired-features = []\ntest = true\n\n[dependencies]\nactix-cors = \"0.5.1\"\nactix-web = \"3.2.0\"\nanyhow = \"1.0.34\"\nbase-62 = \"0.1.1\"\nlazy_static = \"1.4.0\"\nserde = \"1.0.117\"\nserde_derive = \"1.0.117\"\ntokio = \"0.2\"\ntracing = \"0.1.21\"\ntracing-actix-web = \"0.2.1\"\ntracing-bunyan-formatter = \"0.1.6\"\ntracing-futures = \"0.2.4\"\ntracing-log = \"0.1.1\"\ntracing-subscriber = \"0.2.15\"\n\n[dependencies.rusoto_core]\nfeatures = [\"rustls\"]\noptional = false\nversion = \"0.45.0\"\n\n[dependencies.rusoto_s3]\nfeatures = [\"rustls\"]\noptional = false\nversion = \"0.45.0\"\n\n[dependencies.uuid]\nfeatures = [\"v4\"]\noptional = false\nversion = \"0.8.1\"\n\n[lib]\nbench = true\ncrate-type = [\"rlib\"]\ndoc = true\ndoctest = true\nedition = \"2018\"\nharness = true\nname = \"elposhigu\"\npath = \"src/lib.rs\"\nplugin = false\nproc-macro = false\nrequired-features = []\ntest = true\n\n[package]\nauthors = [\"James McMurray <jamesmcm03@gmail.com>\"]\nautobenches = true\nautobins = true\nautoexamples = true\nautotests = true\ncategories = []\ndescription = \"Minimal pastebin service written in Rust + Actix Web\"\nedition = \"2018\"\nhomepage = \"https://github.com/jamesmcm/elposhigu\"\nkeywords = []\nlicense = \"MIT/Apache-2.0\"\nname = \"elposhigu\"\npublish = true\nreadme = \"README.md\"\nrepository = \"https://github.com/jamesmcm/elposhigu\"\nversion = \"0.1.0\"\n"
      }
    ],
    "lock_file": "#..."
}
```

However, `cargo chef cook` generates a `backend/Cargo.toml` with no `default_features` entries:

```toml
[dependencies.rusoto_core]
features = ["rustls"]
optional = false
version = "0.45.0"

[dependencies.rusoto_s3]
features = ["rustls"]
optional = false
version = "0.45.0"
```

In this case, this causes the build to fail due to having both rustls (and native-tls still included):

```
error[E0252]: the name `tls` is defined multiple times
  --> /home/archie/.cargo/registry/src/github.com-1ecc6299db9ec823/rusoto_core-0.45.0/src/lib.rs:25:5
   |
23 | use hyper_rustls as tls;
   |     ------------------- previous import of the module `tls` here
24 | #[cfg(feature = "native-tls")]
25 | use hyper_tls as tls;
   |     ^^^^^^^^^^^^^^^^ `tls` reimported here
   |
   = note: `tls` must be defined only once in the type namespace of this module
help: you can use `as` to change the binding name of the import
   |
25 | use hyper_tls as other_tls;
   |     ^^^^^^^^^^^^^^^^^^^^^^

error: aborting due to previous error

For more information about this error, try `rustc --explain E0252`.
error: could not compile `rusoto_core`

To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
error: build failed
thread 'main' panicked at 'Exited with status code: 101', /home/archie/.cargo/registry/src/github.com-1ecc6299db9ec823/cargo-chef-0.1.8/src/recipe.rs:88:27
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

Even using `--no-default-features` for the backend package directly doesn't seem to help.

Is it possible to use crate-level default features?

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -28,7 +28,7 @@ env_logger = "0.8.1"
 globwalk = "0.8.0"
 anyhow = "1.0.33"
 pathdiff = "0.2.0"
-cargo-manifest = "0.2.0"
+cargo-manifest = "0.2.1"
 fs-err = "2.5.0"
 toml = "0.5.7"
 

